### PR TITLE
Address FIXME comments

### LINKED ISSUE #430: Address FIXME comments for flake8
There are FIXME comments from #381 related to `flake8` issues that were caught and ignored.

### LINKED ISSUE #431: Address FIXME comment for mypy
There are FIXME comments from #381 related to `mypy` issues that were caught and ignored.

### PR DESCRIPTION
**Pull request by forsyth2**
Address FIXME comments. Resolves #430. Resolves #431.

### CODE REVIEW

**Comment by forsyth2** [comment]
@tomvothecoder This is ready for review. I ran the unit tests and the complete run test.

A few `FIXME`s remain; I didn't see a good way to fix them
```
$ git grep -in fixme
acme_diags/acme_diags_driver.py:290:            # FIXME: error: Module has no attribute "run_diag"
acme_diags/plot/__init__.py:24:        # FIXME: error: Module has no attribute "plot"
tests/system/test_diags.py:136:        # FIXME: "Type[TestAllSets]" has no attribute "results_dir"
```

Also, it looks like #434 changed how some of the plots look, so I followed the steps at https://e3sm-project.github.io/e3sm_diags/_build/html/master/dev_guide/testing.html#complete-run-test to update the expected images.

File: acme_diags/container/e3sm_diags_container.py [hunk 1]
```diff
@@ -37,8 +37,7 @@ def run_cmd(cmd):
     """
     print("Using the command: {}".format(cmd))
     # p = subprocess.Popen(cmd, shell=True)
```
Comment (tomvothecoder) [inline acme_diags/container/e3sm_diags_container.py:39]: Do we still need this line if it is commented out?

File: acme_diags/derivations/acme.py [hunk 2]
```diff
@@ -415,6 +401,26 @@ def cosp_bin_sum(
     return cld_bin_sum
 
 
+def determine_tau(tau, tau_low0, tau_high0):
```
Comment (tomvothecoder) [inline acme_diags/derivations/acme.py:404]: I would add unit tests to this new function, similar to the ones here: https://github.com/E3SM-Project/e3sm_diags/blob/master/tests/acme_diags/derivations/test_acme.py

**Review by tomvothecoder — COMMENTED** [review]
@forsyth2 Overall it looks good, just some minor comments.

Thank you for addressing the FIXME comments and performing the refactoring!

**Review by tomvothecoder — APPROVED** [review]

**Pull request merged**